src/pilot/dell_nfv.py

```diff
@@ -0,0 +1,376 @@
+                exit_code, _, std_err = Ssh.execute_command(ip,
+                                                            "pwd",
+                                                            user="heat-admin",
+                                                            )
+                retries -= 1
+                if exit_code != 0:
+                    logger.info("Server with IP: {}"
+                                " is not responsive".format(ip))
+                else:
+                    logger.info("Server with IP: {}"
+                                " is responsive".format(ip))
+                    ssh_success_count += 1
+                    break
+                # Waiting 10s before the next attempt
+                time.sleep(10)
+        if ssh_success_count == len(dell_compute):
+            logger.info("All compute nodes are now responsive. Continuing...")
+        else:
+            logger.error("Failed to reboot the nodes or at least "
+                         "one node failed to get back up")
+            raise Exception("At least one compute node failed to reboot")
+
+    @classmethod
+    def calculate_hostos_cpus(self, number_of_host_os_cpu):
+        try:
+            cpu_count_list = []
+            for node in ConfigOvercloud.nodes:
+                # for every compute node get the corresponding drac credentials
+                # to fetch the cpu details
+                node_uuid = node.uuid
+                node_details = ConfigOvercloud.ironic_client.node.get(
+                    node_uuid)
+                node_type = node_details.properties['capabilities'].split(',')[
+                    0].split(':')[1]
+                if 'compute' not in node_type:
+                    # filter for getting compute node
+                    continue
+                drac_ip, drac_user, drac_password = \
+                    ConfigOvercloud.get_drac_credentail.get_drac_creds(
+                        ConfigOvercloud.ironic_client, node_uuid)
+                stor = client.DRACClient(drac_ip, drac_user, drac_password)
+                # cpu socket information for every compute node
+                sockets = stor.list_cpus()
+                cpu_count = 0
+                for socket in sockets:
+                    if socket.ht_enabled:
+                        cpu_count += socket.cores * 2
+                    else:
+                        raise Exception("Hyperthreading is not enabled in "
+                                        + str(node_uuid))
+                        sys.exit(0)
+                cpu_count_list.append(cpu_count)
+
+            min_cpu_count = min(cpu_count_list)
+            if min_cpu_count not in [40, 48, 56, 64, 72, 128]:
+                raise Exception("CPU count should be one of these"
+                                " values : [40,48,56,64,72,128]"
+                                " But number of cpu is " + str(
+                                    min_cpu_count))
+                sys.exit(0)
```

Comment:
The above line should be removed since it will never be executed due to the raise statement above.

Reply:
Removed